Sources/Regex/VirtualMachine.swift

```diff
@@ -132,6 +145,22 @@ extension RECode: RandomAccessCollection {
   }
 }
 
+extension RECode {
+  public func withMatchLevel(_ level: CharacterClass.MatchLevel) -> RECode {
```

Comment:
Should code instead have a level associated with it? Would it make sense to have one override the other or both?

Reply:
I think that makes sense; in that case we'd need a `default` level, or to make it optional, so that we could distinguish between things like these — the second one should still match at the grapheme cluster level:

```swift
Regex {
    "a"
    .any
    "b"
}.scalarSemantic

Regex {
    "a"
    .any.graphemeClusterSemantic
    "b"
}.scalarSemantic
```